### PR TITLE
The Zariski Topology is compact

### PR DESCRIPTION
**Pull request by tirix**
This adds proofs in 4 independent, but related directions:
- the Zariski topology is compact (`~zarcmp`)
- the preimage of a prime ideal by a ring homomorphism is a prime ideal (`~rhmpreimaprmidl`)
- the property of being a prime element in a ring (`~isrprm`)
- the zero ideal is a prime ideal if and only in an integral domain (`~prmidl0`)

I'm also adding some references to Grothendieck's _Éléments de Géométrie Algébrique_ where relevant.
The next step would be to prove that the induced map on prime spectra is continuous.

### CODE REVIEW

**Review by avekens — APPROVED** [review]
Very interesting results!
Would it be useful to have a definition for zero/trivial rings (maybe `` ZRing = { r e. Ring | ( # ` ( Base ` r ) ) = 1 }``)?

**Comment by avekens** [comment]
PS: `ZRing` is already existing, maybe `0Ring` then...

**Comment by tirix** [comment]
> Would it be useful to have a definition for zero/trivial rings (maybe `` 0Ring = { r e. Ring | ( # ` ( Base ` r ) ) = 1 } ``)?

That would be a useful discussion to launch!

For zero rings, I've followed theorems in the main part: [~0ring](https://us.metamath.org/mpeuni/0ring.html), [~0ring01eq](https://us.metamath.org/mpeuni/0ring01eq.html), [~0ringnnzr](https://us.metamath.org/mpeuni/0ringnnzr.html), which are all in the form ``( ( R e. Ring /\ ( # ` B ) = 1 ) ->  ...``

If I were to choose a "class of zero rings", I'd say `( Ring \ NzRing )` is the best, because it is concise and does not need introducing a new definition. You actually have a few theorems with this form in your mathbox ([~0ringdif](https://us.metamath.org/mpeuni/0ringdif.html) and following)

File: mmset.raw.html [hunk 1]
```diff
@@ -5424,6 +5424,13 @@
 Publishing Company, Reading, MA (1994) [QA39.2.G733].
 </LI>
 <LI>
+<A NAME="EGA"></A> [EGA] Grothendieck, Alexandre and Dieudonn&eacute;,
+Jean <I>&Eacute;l&eacute;ments de g&eacute;om&eacute;trie alg&eacute;brique
+Institut des hautes &eacute;tudes scientifiques, Publ. Math. No. 4 (1960),
+available at <A HREF="http://www.numdam.org/item/PMIHES_1960__4__5_0.pdf">
+NumDam</A>.
```
Comment (icecream17) [inline mmset.raw.html:5431]: unclosed `<I>`

**Review by icecream17 — APPROVED** [review]

**Pull request merged**